tests/installation/ipxe_install.pm

```diff
@@ -75,6 +75,14 @@ sub set_bootscript {
     $console =~ s/^\s+|\s+$//g;
     $mirror_http =~ s/^\s+|\s+$//g;
 
+    # Install other release rather than the VERSION
+    if (get_var('REPO_0_TO_INSTALL')) {
+        my $openqa_url = get_required_var('OPENQA_URL');
+        $openqa_url = 'http://' . $openqa_url unless $openqa_url =~ /http:\/\//;
```

Comment:
This condition will result in `$openqa_url = 'http://https://openqa.opensuse.org'`

Reply:
Yes, good catch. thanks. fixed.

Reply:
The correct fix is this:

    $mirror_http = autoinst_url("/assets/repo/" . get_var("REPO_0_TO_INSTALL"));

Reply:
But more importantly, why do you even add the `REPO_0_TO_INSTALL` variable to do the exact same thing as `MIRROR_HTTP`?

Reply:
> The correct fix is this:
> 
> ```
> $mirror_http = autoinst_url("/assets/repo/" . get_var("REPO_0_TO_INSTALL"));
> ```

Just tried it out, but the path is not correct. It is the job space in worker instead of the http path in openqa repo.

```
https://openqa.suse.de:20523/T8bHv8ZC0urZFd92/assets/repo/fixed/SLE-12-SP5-Server-DVD-x86_64-GM-DVD1

```

I need:
```
https://openqa.suse.de/assets/repo/fixed/SLE-12-SP5-Server-DVD-x86_64-GM-DVD1
```

Reply:
Ah, right. Normal `REPO_N` would be copied to the worker and then accessed via `autoinst_url()`. But the `_TO_INSTALL` suffix prevents that here.

On the other hand, when the worker is physically located far away from the OpenQA server, this approach will stress the network a lot. But we also have several mirror hosts located near workers which are supposed to be used throught the `MIRROR_HTTP` variable. Why not use that?

Reply:
`MIRROR_HTTP` is for installing the current products, fg.sle-15-SP5. but in virt tests, we need install different products, such as sle-12-sp5, sle15-sp4. For these group of tests, `REPO_0_TO_INSTALL` is introduced to flag which product will be installed.

The baremetal machines locate in different locations and VLAN.  medium in openqa server is a good place to access, and it is easier to maintain.

Reply:
> `MIRROR_HTTP` is for installing the current products, fg.sle-15-SP5. but in virt tests, we need install different products, such as sle-12-sp5, sle15-sp4. For these group of tests, `REPO_0_TO_INSTALL` is introduced to flag which product will be installed.

All repos available on OSD are also available on the dist mirrors, e.g. sle12-sp5 installer repo you've used as an example above (SLE-12-SP5-Server-DVD-x86_64-GM-DVD1) would be `http://[nearest dist mirror]/install/SLP/SLE-12-SP5-Server-GM/x86_64/DVD1/`. In the worst case, you can set `MIRROR_HTTP` to the OSD repo URL.

> The baremetal machines locate in different locations and VLAN. medium in openqa server is a good place to access, and it is easier to maintain. `REPO_0_TO_INSTALL=fixed/SLE-xx` has been widely used in virt tests, code change should be compitible with all existing tests.

You can set `MIRROR_HTTP` in the medium like this:

    MIRROR_HTTP=http://%MIRROR_SERVER%/install/SLP/SLE-12-SP5-Server-GM/x86_64/DVD1/

and then set `MIRROR_SERVER` to the preferred mirror hostname on each worker.

Reply:
We cannot set MIRROR_HTTP in the medium for each test suite because they are in same job group and run with same medium. All the tests are triggered at same time with same MIRROR_HTTP in one medium. ie. We install sle12sp5 in a sle15sp6 job group.  As far as I know, MIRROR_HTTP is passed in the command to trigger an entire job group(or even a product), it will override the configurations in test suite level.

Reply:
You can set `+MIRROR_HTTP=...` in the testsuite or job entry in the jobgroup YAML. The `+` will override the setting from medium and scheduled product.

Reply:
See for example the ALP config in https://openqa.suse.de/admin/job_templates/498 where `+HDD_1` is used to override `HDD_1` set by `isos post`.

Reply:
Yes, `+MIRROR_HTTP=` works well in virt tests with non-developing versions. In addition, setting it in test suites is more flexable. fg. http://10.67.129.27/tests/94